src/Header/Header.jsx

```diff
@@ -77,12 +77,48 @@ export class Header extends Component {
 		}
 	}
 
+	dropdownToggleHandler(idOfOpenDropdown) {
+		// If we've got an onDropdownOpen or onDropdownClose prop call the onDropdownOpen or onDropdownClose callback based on idOfOpenDropdown.
+		const { onDropdownOpen, onDropdownClose } = this.props;
+
+		if (onDropdownOpen) {
+			const onDropdownOpenCallback =
```

Comment:
I wonder if it's worth abstracting this logic out for checking for a function/name of a function as it's used in a few places?